Library/Validation/ValidatableModelBase.cs

```diff
@@ -100,6 +103,18 @@ public void MarkAsClean()
         [JsonIgnore]
         public bool IsDirty { get; }
 
+        public virtual bool Set<T>(ref T storage, T value, [CallerMemberName]string propertyName = null, bool validateAfter = true)
```

Comment:
Instead of a Boolean, let's use an enum. Something like this `public enum AfterActions { None, Validate, ValidateAndRevert }`. Make sense? I like where this is going.